RELEASING.md

```diff
@@ -8,11 +8,18 @@ This project does not have a strict release schedule. However, we would make a r
   - No release will be done if no changes got merged to the `master` branch during the above mentioned window.
   - Releases may be done more frequently than the above mentioned window.
 
-## Make Sure Everything Works
-
-- Check and upgrade dependencies.
-- Check the build status on `master` is green.
-- Manually test the changes that will be published.
+## Prerelease checklist
+
+* Check and upgrade dependencies where it applies and makes sense.
+  - Create a distinct pull request and test your changes since it maybe introduce regressions.
+  - While using the latest versions of dependencies is advised, it may not always be possible due to potential compatibility issues.
+  - Upgraded dependencies should be thoroughly considered and tested to ensure they are safe!
+* Make sure tests are passing.
+  - Locally and in the continuous integration system.
+* Manually test changes included in the new release.
```

Comment:
Could we link to a doc like Testing or Development here?

Reply:
I don't think it's needed since the entry point of the doc is the README already. Both the README and DEVELOPMENT doc are needed here.